### PR TITLE
Improve DUST check

### PR DESCRIPTION
**Pull request by isocolsky**
Related issue bitpay/copay#4343

### CODE REVIEW

**Comment by dabura667** [comment]
A large amount of nodes have changed from bitcoind defaults to 5 sats per byte... just FYI.

**Comment by isocolsky** [comment]
@dabura667 BitPay's internal nodes have a 1 sat / byte relay fee and I believe miners would rather use a low value as well. Do you think this could affect the time to confirmation or otherwise have a negative impact on Copay's users?

File: lib/server.js [hunk 1]
```diff
@@ -2013,6 +2027,20 @@ WalletService.prototype.createTx = function(opts, cb) {
         self._selectTxInputs(txp, opts.utxosToExclude, next);
       },
       function(next) {
+        self._getNetworkParams(wallet.network, function(err, params) {
+          if (err) return next(err);
+
+          var minRelayFee = params.minRelayFee || Defaults.MIN_RELAY_FEE_SAT_PER_BYTE;
+          var dustThreshold = Math.round(3 * minRelayFee * txp.getEstimatedSize() * (1 + Defaults.DUST_MARGIN));
```
Comment (dabura667) [inline lib/server.js]: Shouldn't this be:
```javascript
var dustThreshold = Math.round(3 * minRelayFee * 182 * (1 + Defaults.DUST_MARGIN));
```
dust threshold is not a function of any transaction, it is a function of the minRelayTx under the assumption of the smallest possible contribution an output could make to the byte size... (1 34 byte output and 148 byte input... so 182 bytes)

**Comment by dabura667** [comment]
@isocolsky with dust limit and minRelayTxFee you want to assume higher for safety.

With 1 sat/byte minRelayTxFee the dust limit is 546 sats... with 5, it's 2730 sats. That is currently $0.016 USD compared to $0.003 USD...

maybe compensate by making default Defaults.DUST_MARGIN = 1.5; which will essentially make default be * 2.5 of the "supposed" default dust 546 sats.

I would much rather be on the safe side... and micropayments don't exist anyways besides people "testing out" with as small an amount possible... (which is why some people get their transactions stuck.)

File: lib/server.js (hunk 1 above)
Comment (isocolsky) [inline lib/server.js]: Yes, you are correct, my mistake, will fix. Thank you for catching this!

**Comment by matiu** [comment]
ACK!

**Comment by isocolsky** [comment]
Special thanks to @dabura667 for great review & comments!

**Pull request merged**